nengo_loihi/simulator.py

```diff
@@ -306,64 +305,65 @@ def __init__(  # noqa: C901
             assert self.model.dt == dt
 
         self.precompute = precompute
-        self.networks = None
+        self.network = network
         self.sims = OrderedDict()
         self._run_steps = None
 
-        if network is not None:
-            nengo.rc.set("decoder_cache", "enabled", "False")
-            config.add_params(network)
-
-            # ensure seeds are identical to nengo
-            seed_network(network, seeds=self.model.seeds,
-                         seeded=self.model.seeded)
-
-            # split the host into one, two or three networks
-            self.networks = split(
-                network,
-                precompute=precompute,
-                node_neurons=self.model.node_neurons,
-                node_tau=self.model.decode_tau,
-                remove_passthrough=remove_passthrough,
-            )
-            network = self.networks.chip
-
-            self.model.chip2host_params = self.networks.chip2host_params
-
-            self.chip = self.networks.chip
-            self.host = self.networks.host
-            self.host_pre = self.networks.host_pre
-
-            if len(self.host_pre.all_objects) > 0:
-                host_pre_model = self._get_host_model(
-                    self.host_pre, dt=dt, seeds=self.model.seeds,
-                    seeded=self.model.seeded)
-                self.sims["host_pre"] = nengo.Simulator(self.host_pre,
-                                                        dt=self.dt,
-                                                        model=host_pre_model,
-                                                        progress_bar=False,
-                                                        optimize=False)
-
-            if len(self.host.all_objects) > 0:
-                host_model = self._get_host_model(
-                    self.host, dt=dt, seeds=self.model.seeds,
-                    seeded=self.model.seeded)
-                self.sims["host"] = nengo.Simulator(
-                    self.host,
-                    dt=self.dt,
-                    model=host_model,
-                    progress_bar=False,
-                    optimize=False)
-            elif not precompute:
-                # If there is no host and precompute=False, then all objects
-                # must be on the chip, which is precomputable in the sense that
-                # no communication has to happen with the host.
-                # We could warn about this, but we want to avoid people having
-                # to specify `precompute` unless they absolutely have to.
-                self.precompute = True
-
-            # Build the network into the model
-            self.model.build(network)
+        if network is None:
```

Comment:
Why stop allowing `network == None`?

Reply:
We had some discussion about this in nengo slack. This wasn't being used (or tested) anywhere, and---prior to the refactoring---didn't make a whole lot of sense, because, if we look at these lines in the original version:

https://github.com/nengo/nengo-loihi/blob/da2053ff5847fa8c68b4a4d30378c2cead1edb58/nengo_loihi/simulator.py#L313-L366

there are a large number of attributes added to `self` within this block that are required later. So a user would need to do a few very special things in order to get this to work, which might easily change. Exposing this kind of untested surface area for no good reason seemed like a bad idea to me. @tbekolay agreed that it made sense to disallow for now.

After this refactoring is complete, it could make sense to add back in. Currently I think the need for the attribute `self.model.splitter_directive` being filled in is really the only thing holding this back for now. I'd love to have a better way of organizing the way that model parameters are filled in within the simulator / builder.

Reply:
I agree, there's not a whole lot of demand for it and it's not well-tested. It is something that we allow in the reference backend. I think going forward, we need to discuss whether it's something we want to support, and if so, add a `nengo` test for it to make sure backends support it. I've made an issue nengo/nengo#1519.